### ISSUE TITLE
Mac: Record filesystem type (APFS / HFS+) in telemetry stream

### ISSUE THREAD

**Issue by jeschu1** [microsoft/VFSForGit#1134]
Run performance testing using both, and note the differences

**Comment by jrbriggs** [comment]
I think the first step here is to simply record the filesystem type in EnlistmentInfo

**Comment by jeschu1** [comment]
@jrbriggs I also did some analysis using the Mac Large Repro Perf tests

I did two runs on HFS+ and APFS each and compared the results.
`Enumerating Folders` appears to be faster on HFS+ than APFS.  
On Repo 1: 119 versus 137 
On Repo 2: 155 versus 184

Repo 1: {"File placeholder count":910759,"Folder placeholders count":108237}
Repo 2: {"File placeholder count":354853,"Folder placeholders count":147672}

The crux of what Enumerating Folders does, is create places holder files and downloading sizes (this should be the same for both). @nickgra, @wilbaker, @pmj for reference.  As an aside we don't actually need the sizes for Mac, so this may be an area for optimization in the future.

We still of course need to do the telemetry asked for in this ticket.
